src/destructors.md

```diff
@@ -1,28 +1,24 @@
 # Destructors
 
-When an [initialized]&#32;[variable] in Rust goes out of scope or a [temporary]
-is no longer needed its _destructor_ is run. [Assignment] also runs the
-destructor of its left-hand operand, unless it's an uninitialized variable. If a
-[struct] variable has been partially initialized, only its initialized fields
-are dropped.
+When an [initialized]&#32;[variable] or [temporary] in Rust goes out of
+[scope](#drop-scopes) its *destructor* is run, or it is *dropped*. [Assignment]
+also runs the destructor of its left-hand operand, if it's initialized. If a
+variable has been partially initialized, only its initialized fields are
+dropped.
 
-The destructor of a type consists of
+The destructor of a type `T` consists of
 
-1. Calling its [`std::ops::Drop::drop`] method, if it has one.
+1. If `T: Drop`, calling [`<T as std::ops::Drop>::drop`]
 2. Recursively running the destructor of all of its fields.
-    * The fields of a [struct], [tuple] or [enum variant] are dropped in
-      declaration order. \*
-    * The elements of an [array] or owned [slice][array] are dropped from the
-      first element to the last. \*
-    * The captured values of a [closure] are dropped in an unspecified order.
+    * The fields of a [struct] or [tuple] are dropped in declaration order.
+    * The fields of the active [enum variant] are dropped in declaration order.
+    * The elements of an [array] or owned [slice] are dropped from the
```

Comment:
I don't think that it's specified anywhere authoritative, but judging from the comments on that RFC I think that the RFC left it out by accident.

Reply:
cc @nikomatsakis -- I think this looks quite reasonable and justified; can you confirm that this was intended?

Reply:
I believe it was intended, yes. But more generally, if it matches the current behavior, I think we can confirm it, because the meta-intent of RFC #1857 was to specify that built-in types will not change their drop order.